cmd/telemetrygen/internal/traces/config.go

```diff
@@ -36,4 +50,15 @@ func (c *Config) Flags(fs *pflag.FlagSet) {
 	fs.BoolVar(&c.Batch, "batch", true, "Whether to batch traces")
 	fs.IntVar(&c.LoadSize, "size", 0, "Desired minimum size in MB of string data for each trace generated. This can be used to test traces with large payloads, i.e. when testing the OTLP receiver endpoint max receive size.")
 	fs.DurationVar(&c.SpanDuration, "span-duration", 123*time.Microsecond, "The duration of each generated span.")
+
+	// TLS CA configuration
+	fs.StringVar(&c.CaFile, "ca-cert", "", "Trusted Certificate Authority to verify collector receiver certificate")
+
+	// mTLS configuration
+	fs.BoolVar(&c.ClientAuth.Enabled, "mtls", false, "Whether to require client authentication for mTLS")
+	fs.StringVar(&c.ClientAuth.ClientCertFile, "client-cert", "", "Client certificate file")
+	fs.StringVar(&c.ClientAuth.ClientKeyFile, "client-key", "", "Client private key file")
+
+	// Timeout for connection failures
+	fs.DurationVar(&c.TimeOut, "timeout", 60*time.Second, "Timeout setting when testing failure conditions")
```

Comment:
The timeout seems independent from the TLS support, can we have this on a separate PR?

Reply:
Removed.